contentcuration/kolibri_public/views_v1.py

```diff
@@ -91,22 +92,79 @@ def get_channel_name_by_id(request, channel_id):
     return Response(channel_info)
 
 
+device_info_keys = {
+    "1": [
+        "application",
+        "kolibri_version",
+        "instance_id",
+        "device_name",
+        "operating_system",
+    ],
+    "2": [
+        "application",
+        "kolibri_version",
+        "instance_id",
+        "device_name",
+        "operating_system",
+        "subset_of_users_device",
+    ],
+    "3": [
+        "application",
+        "kolibri_version",
+        "instance_id",
+        "device_name",
+        "operating_system",
+        "subset_of_users_device",
+        "min_content_schema_version",
+    ],
+}
+
+DEVICE_INFO_VERSION = "3"
+
+
+def get_device_info(version=DEVICE_INFO_VERSION):
+    """
+    Returns metadata information about the device
+    The default kwarg version should always be the latest
+    version of device info that this function supports.
+    We maintain historic versions for backwards compatibility
+    """
+
+    if version not in device_info_keys:
+        version = DEVICE_INFO_VERSION
```

Comment:
I suppose this could become a footgun when dealing with [caching of the info endpoint](https://github.com/learningequality/studio/issues/3973), say for example someone sends `?v=4` before we deploy a version `4`. Although, it's also a good idea for situations where we add a new version in Kolibri and are late to make the same change in Studio.

If it was moved into the endpoint, we could disable caching in this situation.

Reply:
Ah, good point - I'll add a note in the caching issue!